flow/connectors/clickhouse/clickhouse.go

```diff
@@ -180,12 +209,13 @@ func NewClickHouseConnector(
 	}
 
 	connector := &ClickHouseConnector{
-		database:         database,
-		PostgresMetadata: pgMetadata,
-		config:           config,
-		logger:           logger,
-		credsProvider:    &clickHouseS3CredentialsNew,
-		s3Stage:          NewClickHouseS3Stage(),
+		database:           database,
+		PostgresMetadata:   pgMetadata,
+		dbSupportsExchange: true,
```

Comment:
can do a minversion check? we require a pretty recent version, so can probably assume `true` & drop variable

Reply:
true/false here depends on the used database engine - only atomic supports it, whereas lazy doesn't.

Reply:
however, I can just try to run an exchange and if there's an error just transparently fallback to the old drop/rename method.

Reply:
yes, I think try & fallback on error makes sense. Even better if you can fallback on specific errors